models/qualtrics__distribution.sql

```diff
@@ -57,20 +57,20 @@ pivoted_metrics as (
     group by 1,2
 ),
 
-calc_medians as (
-    
-    select * from (
-        select 
-            distribution_id, 
-            source_relation,
-            {{ fivetran_utils.percentile(percentile_field='time_to_open_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_open_in_seconds,
-            {{ fivetran_utils.percentile(percentile_field='time_to_start_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_start_in_seconds,
-            {{ fivetran_utils.percentile(percentile_field='time_to_complete_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_complete_in_seconds
-
-        from distribution_contact
-        {% if target.type == 'postgres' %} group by 1,2 {% endif %}
-    )
-    {% if target.type != 'postgres' %} group by 1,2,3,4,5 {% endif %}
+calc_medians as ( 
+     
+    select * from ( 
+        select  
+            distribution_id,  
+            source_relation, 
+            {{ fivetran_utils.percentile(percentile_field='time_to_open_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_open_in_seconds, 
+            {{ fivetran_utils.percentile(percentile_field='time_to_start_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_start_in_seconds, 
+            {{ fivetran_utils.percentile(percentile_field='time_to_complete_in_seconds', partition_field='distribution_id,source_relation', percent='0.5') }} as median_time_to_complete_in_seconds 
+ 
+        from distribution_contact 
+        {% if target.type == 'postgres' %} group by 1,2 {% endif %} 
+    ) 
+    {% if target.type != 'postgres' %} group by 1,2,3,4,5 {% endif %} 
```

Comment:
I am not fully following this cte. Would you be able to explain the need for the subquery and why postgres needs to be treated differently?

Reply:
yeah so for postgres, this [macro](https://github.com/fivetran/dbt_fivetran_utils/tree/v0.4.7#percentile-source) uses an aggregate function, whereas it is a window function on other databases.  i have the second `group by` in order to roll up everything to the same grain (essentially mimicking an aggregate function)

Reply:
added some in-line comments

Reply:
Ahhh okay thanks for the explanation and also the inline comments. Thanks!